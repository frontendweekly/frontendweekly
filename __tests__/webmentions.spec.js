const wm = require('../src/_data/webmentions');

describe('webmentions', () => {
  test('it should return expected field', async () => {
    // Act
    const mentions = await wm(); // ?
    // Assert
    expect(mentions[0]).toMatchSnapshot({
      author: {
        name: expect.any(String),
        photo: expect.any(String),
        type: expect.any(String),
        url: expect.any(String),
      },
      content: {
        text: expect.any(String),
      },
      'repost-of': expect.any(String),
      published: expect.anything(),
      type: expect.any(String),
      url: expect.any(String),
      'wm-id': expect.any(Number),
      'wm-private': expect.any(Boolean),
      'wm-property': expect.any(String),
      'wm-received': expect.anything(),
      'wm-source': expect.any(String),
      'wm-target': expect.any(String),
    });
  });
});
