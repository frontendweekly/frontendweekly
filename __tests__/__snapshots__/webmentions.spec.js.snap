// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webmentions it should return expected field 1`] = `
Object {
  "author": Object {
    "name": Any<String>,
    "photo": Any<String>,
    "type": Any<String>,
    "url": Any<String>,
  },
  "content": Object {
    "text": Any<String>,
  },
  "published": Anything,
  "repost-of": Any<String>,
  "type": Any<String>,
  "url": Any<String>,
  "wm-id": Any<Number>,
  "wm-private": Any<Boolean>,
  "wm-property": Any<String>,
  "wm-received": Anything,
  "wm-source": Any<String>,
  "wm-target": Any<String>,
}
`;
